scripts/install_third_party.py

```diff
@@ -165,20 +190,76 @@ def return_json(source_url):
     return json.loads(response)
 
 
+def test_manifest_syntax(dependency_type, dependency_dict):
+    """This checks syntax of the manifest.json dependencies.
+
+    Display warning message when there is an error and terminate the program.
+    Args:
+      dependency_type: dependency download format.
+      dependency_dict: manifest.json dependency dict
+    """
+    keys = dependency_dict.keys()
+    mandatory_keys = DOWNLOAD_FORMATS_TO_MANIFEST_KEYS[
+        dependency_type]['mandatory_keys']
+    optional_key_pairs= DOWNLOAD_FORMATS_TO_MANIFEST_KEYS[
```

Comment:
What is the condition needed for this check to work? Is it that exactly one key from each pair must exist? I would write that in a comment; it's not obvious.